File: backend/src/main/java/com/artgallery/domain/entity/Artwork.java
package com.artgallery.domain.entity;

import com.artgallery.common.BaseEntity;
import com.artgallery.domain.enums.ArtworkStatus;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OrderBy;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "artworks")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Artwork extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "artist_id", nullable = false)
    private ArtistProfile artist;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, unique = true)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String story;

    private String medium;
    private String style;

    @Column(name = "year_created")
    private Integer yearCreated;

    @Column(name = "width_cm")
    private BigDecimal widthCm;

    @Column(name = "height_cm")
    private BigDecimal heightCm;

    @Column(name = "depth_cm")
    private BigDecimal depthCm;

    @Column(nullable = false, precision = 14, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    @Builder.Default
    private String currency = "INR";

    @Column(nullable = false)
    @Builder.Default
    private Integer quantity = 1;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ArtworkStatus status = ArtworkStatus.DRAFT;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(name = "view_count", nullable = false)
    @Builder.Default
    private Long viewCount = 0L;

    @Version
    @Column(nullable = false)
    @Builder.Default
    private Long version = 0L;

    @OneToMany(mappedBy = "artwork", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("sortOrder ASC")
    @Builder.Default
    private List<ArtworkImage> images = new ArrayList<>();
}

File: frontend/src/types/artwork.ts
export interface ArtworkImage {
  id?: string
  storageKey?: string
  originalUrl?: string | null
  thumbnailUrl?: string | null
  mimeType?: string
  width?: number
  height?: number
  sizeBytes?: number
  primary?: boolean
  sortOrder?: number
}

export interface Artwork {
  id: string
  artistId?: string
  artistName?: string
  title: string
  slug?: string
  description?: string | null
  story?: string | null
  medium?: string | null
  style?: string | null
  yearCreated?: number | null
  widthCm?: number | string | null
  heightCm?: number | string | null
  depthCm?: number | string | null
  price?: number | string | null
  currency?: string | null
  quantity?: number | null
  status?: string
  categoryId?: string | null
  categoryName?: string | null
  viewCount?: number
  createdAt?: string
  updatedAt?: string
  images?: ArtworkImage[]
}

export interface ArtworkCreateRequest {
  title: string
  description?: string
  story?: string
  medium?: string
  style?: string
  yearCreated?: number
  widthCm?: number
  heightCm?: number
  depthCm?: number
  price: number
  currency?: string
  quantity?: number
  categoryId?: string
}

export interface ArtworkUpdateRequest extends Partial<ArtworkCreateRequest> {}

export interface AttachImagesRequest {
  storageKeys: string[]
  primaryKey?: string
}

export interface ArtworkSearchParams {
  q?: string
  category?: string
  medium?: string
  minPrice?: number
  maxPrice?: number
  artistId?: string
  sort?: string
  page?: number
  size?: number
}

export interface Category {
  id: string
  name: string
  slug?: string
  description?: string | null
  parentId?: string | null
}

export interface MediaUpload {
  storageKey: string
  originalUrl?: string
  thumbnailUrl?: string
  mimeType?: string
  sizeBytes?: number
}
